js/components/src/i18n/i18next-config.ts

```diff
@@ -0,0 +1,275 @@
+
+  if (!code) return manifest.fallbackChain;
+
+  // Regional fallbacks
+  if (code.match(/^es-/)) {
+    fallbacks.push("es-ES"); // Spanish fallback
+  } else if (code.match(/^fr-/)) {
+    fallbacks.push("fr-FR"); // French fallback
+  } else if (code.match(/^pt-/)) {
+    fallbacks.push("pt-BR"); // Portuguese fallback
+  } else if (code.match(/^zh-/)) {
+    fallbacks.push("zh-Hant"); // Chinese fallback
+  }
+
+  // Add manifest fallback chain
+  return [...fallbacks, ...manifest.fallbackChain];
+}
+
+const LOCALE = getCurrentLocale();
+
+export const I18NEXT_CONFIG = {
+  lng: LOCALE,
+  interpolation: {
+    escapeValue: false, // React already safes from XSS
+  },
+  react: {
+    useSuspense: false, // Prevent Android crashes
+  },
+  i18nFormat: {
+    fluentBundleOptions: {
+      useIsolating: false,
+      functions: {
+        VOWORCON: ([txt]: [string]) =>
+          "aeiou".indexOf(txt[0].toLowerCase()) >= 0 ? "vow" : "con",
+        JOIN: (args: string[], opts: { separator?: string } = {}) =>
+          args
+            .filter(Boolean)
+            .filter((s) => typeof s === "string")
+            .join(opts.separator || ""),
+      },
+    },
+  },
+  fallbackLng: getFallbackChain,
+  supportedLngs: [
+    ...manifest.supportedLocales,
+    // Include base language codes for i18next fallback matching
+    "en",
+    "pt",
+    "es",
+    "zh",
+    "fr",
+  ],
+  debug: process.env.NODE_ENV === "development",
+};
+
+// Translation loading function that loads compiled JSON files
+async function loadTranslationData(locale: string): Promise<any> {
+  try {
+    if (!manifest.supportedLocales.includes(locale)) {
+      throw new Error(`Unsupported locale: ${locale}`);
+    }
+
+    let translations: any = {};
+
+    try {
+      // For web environments, load from public directory
+      if (typeof window !== "undefined") {
+        const response = await fetch(`/locales/${locale}/messages.json`);
+        if (!response.ok) {
+          throw new Error(`HTTP ${response.status}`);
+        }
+        translations = await response.json();
+      } else {
+        // For React Native, use static requires for bundler compatibility
+        switch (locale) {
+          case "en-US":
+            translations = require("../../public/locales/en-US/messages.json");
+            break;
+          case "pt-BR":
+            translations = require("../../public/locales/pt-BR/messages.json");
+            break;
+          case "es-ES":
+            translations = require("../../public/locales/es-ES/messages.json");
+            break;
+          case "zh-Hant":
+            translations = require("../../public/locales/zh-Hant/messages.json");
+            break;
+          case "fr-FR":
+            translations = require("../../public/locales/fr-FR/messages.json");
+            break;
+          default:
+            throw new Error(`No static translation file for locale: ${locale}`);
```

Comment:
I was about to tell you that this wouldn't work, 'cause the compiler isn't smart enough to not bundle these at build-time just because they're in the else block of `if (typeof window !== "undefined") {`. But... it totally is, they don't get bundled on web. Huh. Good job, Metro.